packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2719,21 +2729,31 @@ mixin RenderObjectWithChildMixin<ChildType extends RenderObject> on RenderObject
   bool debugValidateChild(RenderObject child) {
     assert(() {
       if (child is! ChildType) {
-        throw FlutterError(
-          'A $runtimeType expected a child of type $ChildType but received a '
-          'child of type ${child.runtimeType}.\n'
-          'RenderObjects expect specific types of children because they '
-          'coordinate with their children during layout and paint. For '
-          'example, a RenderSliver cannot be the child of a RenderBox because '
-          'a RenderSliver does not understand the RenderBox layout protocol.\n'
-          '\n'
-          'The $runtimeType that expected a $ChildType child was created by:\n'
-          '  $debugCreator\n'
-          '\n'
-          'The ${child.runtimeType} that did not match the expected child type '
-          'was created by:\n'
-          '  ${child.debugCreator}\n'
-        );
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary(
+            'A $runtimeType expected a child of type $ChildType but received a '
+            'child of type ${child.runtimeType}.'
+          ),
+          ErrorDescription(
+            'RenderObjects expect specific types of children because they '
+            'coordinate with their children during layout and paint. For '
+            'example, a RenderSliver cannot be the child of a RenderBox because '
+            'a RenderSliver does not understand the RenderBox layout protocol.',
+          ),
+          ErrorDescription(''),
```

Comment:
These `ErrorDescription('')` look kinda odd...

Reply:
Should there maybe be a configuration option somewhere to mark paragraphs that should be surrounded by blank lines?

Reply:
Or a specific Error DiagnosticsNode class that just represents a blank line?

Reply:
I agree those look bad.
I'm happy to add an DiagnosticSpacer or ErrorSpacer class that is sugar for this.
Let me know if that sounds reasonable.

Reply:
One other option would be to add a trailing '\n' to the previous ErrorDescription but that makes the intent less obvious to someone reading the message at it would be reasonable to assume that all ErrorDescription objects should just have a trailing `\n`.

Reply:
I like the syntactic sugar idea.

Reply:
Done. Added ErrorSpacer, used it everywhere we were creating empty DiagnosticsNode objects as spacer or adhoc adding multiple line breaks at the end of sections in error messages.
Fixed a couple cases where we had accidentally extra unintended whitespace in the process and added tests.